lib/tpm2_options.c

```diff
@@ -200,24 +203,93 @@ static tcti_conf tcti_get_config(const char *optstr) {
 
     /* Case A */
     if (!optstr[0] && !split[1]) {
-        return conf;
+        return;
     }
 
     /* Case B */
     if (!optstr[0]) {
-        conf.opts = &split[1];
-        return conf;
+        conf->opts = strdup(&split[1]);
+        return;
     }
 
     /* Case C */
     if (!split[1]) {
-        conf.name = fixup_name(optstr);
-        return conf;
+        conf->name = strdup(fixup_name(optstr));
+        return;
     }
 
     /* Case D */
-    conf.name = fixup_name(optstr);
-    conf.opts = &split[1];
+    conf->name = strdup(fixup_name(optstr));
+    conf->opts = strdup(&split[1]);
+    return;
+}
+
+static char* parse_device_tcti(void) {
+    const char *device = getenv(TPM2TOOLS_ENV_DEVICE);
+    return strdup(device);
+}
+
+static char* parse_socket_tcti(void) {
+
+    /*
+     * tpm2_tcti_ldr_load() expects conf->opts to be of the format
+     * "host=localhost,port=2321" for the mssim tcti
+     * 
+     * Max IPV6 IP address, 45 characters   (45)
+     * Ports are 16bit int, 5 characters    (5)
+     * "host=", 5 characters                (5)
+     * "port=", 5 characters                (5)
+     * strlen = 60
+     */
+    size_t optlen = 60;
+    const char *host, *port;
```

Comment:
we dont do N variable declarations on a line.

Reply:
got it